docs/troubleshooting.md

```diff
@@ -25,11 +53,25 @@ If `MicrosoftInstrumentationEngine_FileLogPath` is set along with `MicrosoftInst
 occur. If the FilePath is set to a directory, then an automatically generated `%TEMP%\ProfilerLog_[PID].txt` file is used, where [PID]
 represents the process id and the %TEMP% variable is evaluated against the user account running the process.
 
+#### Instrumentation Method Logging
+
+CLRIE's logging infrastructure is outlined [here](./logging.md). During ICorProfilerCallback::Initialize(), InstrumentationMethods can obtain an InstrumentationMethod-specific logger via `IProfilerManager::GetLoggingInstance()` which will prepend log messages with the `[IM:GUID]`. In addition, IMs can log at a different log level from CLRIE by setting an additional environment variable, allowing IMs to filter out CLRIE or other IM logs. See [ProfilerManagerForInstrumentationMethod](./logging.md#profilermanagerforinstrumentationmethod-hcpp) section for more details.
+
+Log parsing or an event collector can utilize these patterns to track InstrumentationMethod behaviors and potentially help diagnose issues.
+
 ### Debugging the Instrumentation Engine
 
-When using a local build of the Instrumentation Engine, you will need to disable signing validations and checks by setting the
+When using a local build of the Instrumentation Engine or testing your InstrumentationMethod, you will need to disable signing validations and checks by setting the
 `MicrosoftInstrumentationEngine_DisableCodeSignatureValidation` to 1.
 
-Since the CLR Instrumentation Engine initializes on process start, in order for breakpoints in the initialization logic to hit, you will need
-to set the `MicrosoftInstrumentationEngine_DebugWait` environment variable with the value 1. This will cause the Instrumentation Engine to wait
-on startup for a debugger attach event.
\ No newline at end of file
+Since the CLR Instrumentation Engine initializes on process start, in order for breakpoints in the initialization logic to hit, you will need to set the `MicrosoftInstrumentationEngine_DebugWait` environment variable with the value of 1. This will cause the Instrumentation Engine to wait on startup for a debugger attach event.
+
+#### Breakpoint Locations
+
+The best place to start debugging is in [ProfilerManager.cpp](../src/InstrumentationEngine/ProfilerManager.cpp). ProfilerManager is CLRIE's ICorProfiler implementation which gets the first-chance callback during profiling. It controls setting up InstrumentationMethods and forwarding ICorProfiler callback events to them.
+
+Setting a breakpoint inside `CProfilerManager::Initialize()` provides the earliest point where CLRIE is beginning to initialize in the process.
+
+`CProfilerManager::AddInstrumentationMethod()` provides the logic for loading each InstrumentationMethod and calling Initialize() on them.
+
+CLRIE's main purpose is to coordinate instrumentation, so callbacks such as `CProfilerManager::JITCompilationStarted()` and `CProfilerManager::GetReJITParameters()` are also good places to inspect.
```

Comment:
I'm wondering if we can develop a CLR debugging workflow based on public symbol servers for CLR? This way, users can try to debug their il failures at the point of failure (e.g. ProgramInvalidException).

Reply:
I can at least add some details about public symbol servers.